README
Gtk::ImageViewer version 0.03
=============================

The Gtk::ImageViewer module allows a perl developer to use the GtkImageViewer
widget with Gtk2-Perl.  

INSTALLATION

To install this module type the following:

   perl Makefile.PL
   make
   make test
   make install


To avoid installing to a system directory, since this is a beta release,
you can change the installation prefix at Makefile.PL time with

   perl Makefile.PL PREFIX=/some/other/place

This will install the module to the subdirectory lib/perl5 under the given
prefix.  If this is not already in perl's include path, you'll need to tell
perl how to get to this library directory so you can use it; there are three
ways:

  in your environment (the easiest):
    PERL5LIB=/some/other/place/lib/perl5/site_perl:$PERL5LIB

  on the perl command line:
    perl -I /some/other/place/lib/perl5/site_perl yourscript

  in the code of your perl script:
    use lib '/some/other/place/lib/perl5/site_perl';
  

DEPENDENCIES

This module requires these other modules and libraries:

  perl >= 5.8.0
  Glib >= 1.00rc2 (perl module and requisite C libraries)
  Gtk2 >= 1.00rc2 (perl module and requisite C libraries)
  GtkImageViewer >= 0.5 

In order to build it from source, you'll also need

  ExtUtils::Depends >= 0.1
  ExtUtils::PkgConfig >= 1.00


COPYRIGHT AND LICENCE

Copyright (C) 2003 by the gtk2-perl team (see the file AUTHORS for a
complete list)

This library is free software; you can redistribute it and/or
modify it under the terms of the GNU Library General Public
License as published by the Free Software Foundation; either
version 2 of the License, or (at your option) any later version.

This library is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
Library General Public License for more details.

You should have received a copy of the GNU Library General Public
License along with this library; if not, write to the 
Free Software Foundation, Inc., 59 Temple Place - Suite 330, 
Boston, MA  02111-1307  USA.




ChangeLog
2007-02-11  Dov Grobgeld  <dov.grobgeld@gmail.com>

	* Added a few more bindings to functions that have been added
	during the years.




ImageViewer.pm
package Gtk2::ImageViewer;

use 5.008;
use strict;
use warnings;

use Gtk2;

require DynaLoader;

our @ISA = qw(DynaLoader);

our $VERSION = '0.03';

sub dl_load_flags { 0x01 }

bootstrap Gtk2::ImageViewer $VERSION;

# Preloaded methods go here.

1;
__END__
# Below is stub documentation for your module. You'd better edit it!

=head1 NAME

Gtk2::ImageViewer - Perl interface to the GtkImageViewer

=head1 SYNOPSIS

  use Gtk2::ImageViewer;
  Gtk2->init;

  $window = Gtk2::Window->new();

  $imgv = new_from_file Gtk2::ImageViewer(shift);
  $window->add($imgv);

  $window->show_all();

main Gtk2;

  $imgv = new_from_file Gtk2::ImageViewer($filename);
  
  Gtk2->main;

=head1 ABSTRACT

  Perl bindings to the 2.x series of the Gtk Image Viewer widget.

=head1 DESCRIPTION

The Gtk2::ImageViewer module allows a perl developer to use the Gtk
Image Viewer Widget. The Gtk ImageViewer widget is an interactive
image viewer. See http://imagic.weizmann.ac.il/~dov/freesw/gtk/gtk-image-viewer
for an explanation.

To discuss gtk2-perl, ask questions and flame/praise the authors,
join gtk-perl-list@gnome.org at lists.gnome.org.

=head1 SEE ALSO

perl(1), Glib(3pm), Gtk2(3pm)

=head1 AUTHOR

Dov Grobgeld E<lt>dov@imagic.weizmann.ac.ilE<gt>

=head1 COPYRIGHT AND LICENSE

Copyright 2003 by Dov Grobgeld.

This library is free software; you can redistribute it and/or
modify it under the terms of the GNU Library General Public
License as published by the Free Software Foundation; either
version 2 of the License, or (at your option) any later version.

This library is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
Library General Public License for more details.

You should have received a copy of the GNU Library General Public
License along with this library; if not, write to the 
Free Software Foundation, Inc., 59 Temple Place - Suite 330, 
Boston, MA  02111-1307  USA.

=cut


Makefile.PL
use 5.008;
use ExtUtils::MakeMaker;
use Cwd;
use File::Spec;

# minimum required version of dependancies we need to build
our %build_reqs = (
	'perl-ExtUtils-Depends'   => '0.1',
	'perl-ExtUtils-PkgConfig' => '0.1',
	'perl-Glib'               => '1.03',
	'perl-Gtk2'               => '1.00',
	'GtkImageViewer'          => '1.0',
);

# minimum required version of dependancies we need to run
our %runtime_reqs = (
	'GtkImageViewer' => undef,
);

# Writing a fake Makefile ensures that CPAN will pick up the correct
# dependencies and install them.
unless (eval "use ExtUtils::Depends;"
           . "use ExtUtils::PkgConfig;"
           . "use Gtk2::CodeGen;"
           # just seeing if Glib is available isn't enough, make sure
           # it's recent enough, too 
           . "use Glib '$build_reqs{'perl-Glib'}';"
           . "use Gtk2 '$build_reqs{'perl-Gtk2'}';"
           . "1") {
   warn "$@\n";
   WriteMakefile(
         PREREQ_FATAL => 1,
         PREREQ_PM    => {
            Glib::                => $build_reqs{'perl-Glib'},
            Gtk2::                => $build_reqs{'perl-Gtk2'},
            ExtUtils::Depends::   => $build_reqs{'perl-ExtUtils-Depends'},
            ExtUtils::PkgConfig:: => $build_reqs{'perl-ExtUtils-PkgConfig'},
         },
   );
   exit 1; # not reached
}

%pkgcfg = ExtUtils::PkgConfig->find ('gtkimageviewer');

$runtime_reqs{GtkImageViewer} = $pkgcfg{modversion};

mkdir 'build', 0777;

#
# autogeneration
#
Gtk2::CodeGen->parse_maps ('gtkimageviewerperl');
Gtk2::CodeGen->write_boot;

# now we're ready to start creating the makefile.
# we need to use ExtUtils::Depends to get relevant information out of
# the Glib extension, and to save config information for other modules which
# will chain from this one.

$giv2 = ExtUtils::Depends->new ('Gtk2::ImageViewer', 'Gtk2', 'Glib');
$giv2->set_inc ($pkgcfg{cflags});
$giv2->set_libs ($pkgcfg{libs});
$giv2->add_xs (<*.xs>);
$giv2->add_pm ('ImageViewer.pm' => '$(INST_LIBDIR)/ImageViewer.pm');
my $cwd = cwd();
$giv2->add_typemaps (map {File::Spec->catfile($cwd,$_)} 'build/gtkimageviewerperl.typemap');
$giv2->add_headers ('gtkimageviewerperl.h');

$giv2->install (qw(build/gtkimageviewerperl-autogen.h));
$giv2->save_config ('build/IFiles.pm');

WriteMakefile(
    NAME            => 'Gtk2::ImageViewer',
    VERSION_FROM    => 'ImageViewer.pm', # finds $VERSION
    ABSTRACT_FROM   => 'ImageViewer.pm', # retrieve abstract from module
    XSPROTOARG      => '-noprototypes',
    $giv2->get_makefile_vars,
);

package MY;

sub postamble
{
	"
realclean ::
	-rm -Rf build
"
}
package MAIN;
